sdk/src/test/java/com/mapbox/maps/MapSurfaceTest.kt

```diff
@@ -0,0 +1,215 @@
+package com.mapbox.maps
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.view.MotionEvent
+import android.view.Surface
+import com.mapbox.maps.plugin.MapPlugin
+import com.mapbox.maps.renderer.MapboxSurfaceRenderer
+import com.mapbox.maps.renderer.OnFpsChangedListener
+import com.mapbox.maps.renderer.RendererSetupErrorListener
+import com.mapbox.maps.renderer.widget.BitmapWidget
+import io.mockk.*
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNull
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class MapSurfaceTest {
+
+  private lateinit var context: Context
+  private lateinit var mapController: MapController
+  private lateinit var mapSurface: MapSurface
+  private lateinit var mapInitOptions: MapInitOptions
+  private lateinit var mapboxSurfaceRenderer: MapboxSurfaceRenderer
+  private lateinit var surface: Surface
+
+  @Before
+  fun setUp() {
+    context = mockk(relaxUnitFun = true)
+    mapInitOptions = mockk(relaxUnitFun = true)
+    mapController = mockk(relaxUnitFun = true)
+    mapboxSurfaceRenderer = mockk(relaxUnitFun = true)
+    surface = mockk(relaxed = true)
+
+    mapSurface = MapSurface(
+      context,
+      surface,
+      mapInitOptions,
+      mapboxSurfaceRenderer,
+      mapController
+    )
+
+    mockkStatic("com.mapbox.maps.MapboxLogger")
+    every { logI(any(), any()) } just Runs
+  }
+
+  @After
+  fun cleanup() {
+    unmockkStatic("com.mapbox.maps.MapboxLogger")
+  }
+
+  @Test
+  fun testGetSurface() {
+    assertEquals(surface, mapSurface.surface)
+  }
+
+  @Test
+  fun testSurfaceCreated() {
+    every { context.getSystemService(Context.WINDOW_SERVICE) } returns null
+    mapSurface.surfaceCreated()
+    verify { mapboxSurfaceRenderer.surfaceCreated() }
```

Comment:
imo here and everywhere below it makes sense to check `verify(exactly = 1)` or use `verifyOnce` extension (although it may not be shared across tests but some tests use it).

Reply:
Good idea